### PR TITLE
Stop PC failing when completion point outside file

### PR DESCRIPTION
**Pull request by rorygraves**

### CODE REVIEW

**Comment by rorygraves** [comment]
I think this is an unrelated failure - possibly due to the snapshot releases of the refactoring library?  @fommil 

Btw - PR is easier to read in this form: https://github.com/ensime/ensime-server/pull/1323/files?w=1

File: core/src/main/scala/org/ensime/core/Completion.scala [hunk 1]
```diff
@@ -349,9 +357,10 @@ object CompletionUtil {
   def matchesPrefix(m: String, prefix: String,
     matchEntire: Boolean, caseSens: Boolean): Boolean = {
     val prefixUpper = prefix.toUpperCase
-    ((matchEntire && m == prefix) ||
+
+    (matchEntire && m == prefix) ||
```
Comment (rorygraves) [inline core/src/main/scala/org/ensime/core/Completion.scala]: Removal of bonus brackets.

File: core/src/main/scala/org/ensime/core/Completion.scala [hunk 2]
```diff
@@ -312,8 +321,9 @@ object Keywords {
     "yield"
   )
 
-  val keywordCompletions = keywords map { CompletionInfo(_, CompletionSignature(List(), "", false), false, 100, None) }
-
+  val keywordCompletions = keywords map {
```
Comment (rorygraves) [inline core/src/main/scala/org/ensime/core/Completion.scala]: Relayout only

File: core/src/main/scala/org/ensime/core/Completion.scala [hunk 3]
```diff
@@ -77,87 +76,97 @@ trait CompletionControl {
   import CompletionUtil._
 
   def completionsAt(inputP: Position, maxResultsArg: Int, caseSens: Boolean): CompletionInfoList = {
-    val maxResults = if (maxResultsArg == 0) Int.MaxValue else maxResultsArg
+    val origContents = inputP.source.content
+    val point = inputP.endOrCursor
+
+    if (point > origContents.length) {
```
Comment (rorygraves) [inline core/src/main/scala/org/ensime/core/Completion.scala]: 90% of the change is the same code - I've wrapped it in a defensive if block.

**Comment by fommil** [comment]
I see what you mean. Do you mind pushing a small commit to address the changes in the refactoring library, unless you think it's just wrong. Looks small.

**Comment by rorygraves** [comment]
I will do some more tests on that first- but yeah

**Comment by fommil** [comment]
I'm not too keen on the refactoring-library change. It is making things come out as

```scala
   package org.ensime.testing

   import java.lang.Integer
   trait Temp {
     valueOf(5)
     vo("5")
     toBinaryString(27)
   }
```

```scala
   package org.ensime.testing

   import java.lang.Integer
   
   trait Temp {
     valueOf(5)
     vo("5")
     toBinaryString(27)
   }
```

instead of 

```

```

**Comment by rorygraves** [comment]
Yeah, looks like a regression on the library or some weirdness do to the recent fun with repositories on their side?

**Comment by fommil** [comment]
@sschaef @mlangc any ideas what could have caused the above regression?

**Comment by fommil** [comment]
it's got to be https://github.com/scala-ide/scala-refactoring/pull/139

**Comment by fommil** [comment]
ok, it's intentional.

File: core/src/main/scala/org/ensime/core/Completion.scala (hunk 1 above)
Comment (fommil) [inline core/src/main/scala/org/ensime/core/Completion.scala]: oh, excellent. We need a few more in ensime-emacs.

**Comment by sschaef** [comment]
The new behavior makes the reafactoring library safer to use. Before merging, we should have verified first if it breaks your test suite. Sorry about that!

The plan is to let organize imports do the cleanups, i.g. the insertion of newlines if they are necessary.

**Comment by rorygraves** [comment]
@fommil AppVeyor has failed for an unrelated slf4j initialisation issue.

**Comment by fommil** [comment]
Pull requests that have a failing status can’t be merged from a phone.

Ha! Nice feature

**Pull request merged**